system/jlib/jthread.cpp

```diff
@@ -475,22 +492,10 @@ bool Thread::join(unsigned timeout)
         stopped.signal();
         return true;
     }
-    unsigned st = 0;
-    for (;;) {                                              // this is to prevent race with destroy
-                                                        // (because Thread objects are not always link counted!)
-        {
-            CriticalBlock block(ThreadDestroyListLock);
-            if (ThreadDestroyList.find(*this)==NotFound)
-                break;
-        }
-#ifdef _DEBUG
-        if (st==10)     
-            PROGLOG("Thread::join race");   
-#endif
-        Sleep(st);      // switch back to exiting thread (not very elegant!)
-        st++;
-        if (st>10)
-            st = 10; // note must be non-zero for high priority threads
+
+    {
+        CriticalBlock block(ThreadDestroyLock);
```

Comment:
Do we need this CB here ?

Reply:
yes - the comment should probably precede the critical block.  It is preventing the other thread stalling between the signal and the release, and this thread continuing, and disposing of the object that contains the threading object.  (Which does happen...)

Reply:
moved the comment, and clarified it.